src/main/java/seedu/address/model/VersionedAddressBook.java

```diff
@@ -15,31 +15,63 @@ public class VersionedAddressBook extends AddressBook {
     // Keeps track of the current version of AddressBook
     private int currentStatePointer;
 
+    // Maximum number of snapshots we take
+    private int sizeLimit;
+
     // Stores the different versions of AddressBook
     private ArrayList<ReadOnlyAddressBook> addressBookStateList = new ArrayList<>();
 
     /**
      * Creates an instance of VersionedAddressBook.
+     *
      * @param addressBook AddressBook to be initialised as the first state
      */
     public VersionedAddressBook(ReadOnlyAddressBook addressBook) {
         this.currentStatePointer = 0;
+        this.sizeLimit = 20; // Default value
         addressBookStateList.add(this.currentStatePointer, addressBook);
     }
 
+    /**
+     * Creates an instance of VersionedAddressBook.
+     *
+     * @param addressBook AddressBook to be initialised as the first state
+     * @param stepLimit   maximum number of previous states tracked
+     */
+    public VersionedAddressBook(ReadOnlyAddressBook addressBook, int stepLimit) {
+        this.currentStatePointer = 0;
+        this.sizeLimit = stepLimit; // Default value
+        addressBookStateList.add(this.currentStatePointer, addressBook);
+    }
```

Comment:
Wondering when will this command be called? Do we have to create a new command to allow users to set the limit for the number of states we want to track? Since the current implementation is that it is called/initialised under ModelManager when ModelManager is initialised.